app/src/main/java/de/westnordost/streetcomplete/ktx/Resources.kt

```diff
@@ -6,15 +6,23 @@ import android.graphics.drawable.BitmapDrawable
 import android.graphics.drawable.Drawable
 import androidx.annotation.DrawableRes
 import androidx.annotation.RawRes
-import com.esotericsoftware.yamlbeans.YamlReader
+import com.charleskorn.kaml.Yaml
 import de.westnordost.streetcomplete.view.DrawableImage
 import de.westnordost.streetcomplete.view.Image
 import de.westnordost.streetcomplete.view.ResImage
-import java.io.BufferedReader
-import java.io.InputStreamReader
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.builtins.serializer
 
-inline fun <reified T> Resources.getYamlObject(@RawRes id: Int): T =
-    YamlReader(BufferedReader(InputStreamReader(openRawResource(id)))).read(T::class.java)
+inline fun <reified T> Resources.getYamlObject(serializer: KSerializer<T>, @RawRes id: Int): T =
+    Yaml.default.decodeFromStream(serializer, openRawResource(id))
+
+fun Resources.getYamlStringMap(@RawRes id: Int): Map<String, String> =
+    this.getYamlObject(MapSerializer(String.serializer(), String.serializer()), id)
+
+fun Resources.getYamlStringList(@RawRes id: Int): List<String> =
+    this.getYamlObject(ListSerializer(String.serializer()), id)
```

Comment:
It is weird that kaml/kotlinx.serialization doesn't work with generics. For kotlinx.serialization JSON this was no issue. Is this a kaml limitation?

There is also 
https://github.com/Kotlin/kotlinx.serialization/tree/master/formats#yaml-multiplatform
which we might want to use some day anyway (because multiplatform) if not all places where YAML is parsed is in the same place as accessing Android resources.

Reply:
> It is weird that kaml/kotlinx.serialization doesn't work with generics

It should work fine (e.g. `List<LengthUnit>` in `CountryInfo` does serialize just fine), but we need a top-level declaration here. And since there is no `data class` for which it can generate a serializer at compile time, we have to manually construct the serializer ourselves: https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/serializers.md#constructing-collection-serializers

This would be the same for JSON, so it's not a kaml limitation.

Reply:
> There is also
[Kotlin/kotlinx.serialization@master/formats#yaml-multiplatform](https://github.com/Kotlin/kotlinx.serialization/tree/master/formats?rgh-link-date=2022-02-19T13%3A29%3A04Z#yaml-multiplatform) […]
> Edit: But it doesn't support multiline strings yet and at least for the changelog, we use them.

So I'd say let's stick to kaml for now; it should be quite easy to switch to another kotlinx.serialization YAML plugin, because the serializers could stay the same.

Reply:
I found a way to simplify this, see https://github.com/streetcomplete/StreetComplete/commit/adf9a7890076bfcf12862b2ad87e182a8ba11722 :tada: